File: src/main/java/com/gtladd/gtladditions/api/gui/widget/SafeIntInputWidget.java
package com.gtladd.gtladditions.api.gui.widget;

import com.gregtechceu.gtceu.api.gui.widget.IntInputWidget;

import com.google.common.primitives.Ints;

import java.util.function.Consumer;
import java.util.function.Supplier;

public class SafeIntInputWidget extends IntInputWidget {

    public SafeIntInputWidget(Supplier<Integer> valueSupplier, Consumer<Integer> onChanged) {
        super(valueSupplier, onChanged);
    }

    @Override
    protected Integer add(Integer a, Integer b) {
        return Ints.saturatedCast((long) a + b);
    }
}

File: src/main/java/com/gtladd/gtladditions/api/gui/widget/SafeIntInputWidget.kt
package com.gtladd.gtladditions.api.gui.widget

import com.google.common.primitives.Ints
import com.gregtechceu.gtceu.api.gui.widget.IntInputWidget
import java.util.function.Consumer
import java.util.function.Supplier

class SafeIntInputWidget(valueSupplier: Supplier<Int?>?, onChanged: Consumer<Int?>?) :
    IntInputWidget(valueSupplier, onChanged) {
    override fun add(a: Int?, b: Int): Int = Ints.saturatedCast(a!!.toLong() + b)
}
